Makefile

```diff
@@ -1,15 +1,178 @@
-CXXFLAGS=-Oz -fno-stack-protector -ffunction-sections -fdata-sections -Wl,--gc-sections -fno-unwind-tables -fno-asynchronous-unwind-tables -fno-math-errno -ffast-math -fno-unroll-loops -fmerge-all-constants -fno-ident -ffast-math  -Wl,--hash-style=gnu -Wl,--build-id=none -fno-rtti -fno-exceptions -fomit-frame-pointer -fno-unroll-loops -fno-profile-use
-
-.PHONY: demo
-demo: main.cc
-	clang++ `pkg-config --cflags --libs glfw3 gl` ${CXXFLAGS} main.cc -o demo
-	strip -S \
-	  --strip-unneeded \
-	  --remove-section=.note.gnu.gold-version \
-	  --remove-section=.comment \
-	  demo
-	du -b demo | awk '{ print  (65536 - $$1 )} $$1 > 65536 { exit 1 }'
-
-.PHONY: clean
+# CMAKE generated file: DO NOT EDIT!
```

Comment:
When CMake generates it, the makefile does not need to be under source control. Also, you can have CMake generate [Ninja files](https://ninja-build.org/) instead of makefiles. For this project it probably does not matter, but for large projects Ninja is a bit faster.